From f900449e00f7452114084b5c0fafcd6261fc3be4 Mon Sep 17 00:00:00 2001
From: Nhat Thieu <nhat.thieu.xr@renesas.com>
Date: Wed, 29 Sep 2021 09:51:24 +0700
Subject: [PATCH 09/10] Wrong initialize value of clip

Fix initialize value of clip in case setting ISU_CSC_RAW

Signed-off-by: Nhat Thieu <nhat.thieu.xr@renesas.com>
---
 vspm-module/files/vspm/drv/isu/isu_drv_par.c | 9 ++++-----
 1 file changed, 4 insertions(+), 5 deletions(-)

diff --git a/vspm-module/files/vspm/drv/isu/isu_drv_par.c b/vspm-module/files/vspm/drv/isu/isu_drv_par.c
index 20ffc36..97acd61 100755
--- a/vspm-module/files/vspm/drv/isu/isu_drv_par.c
+++ b/vspm-module/files/vspm/drv/isu/isu_drv_par.c
@@ -317,15 +317,14 @@ static long isu_ins_check_wpf_param( struct isu_ch_info *ch_info,
 	wpf_info->swap = (unsigned int)(dst_par->swap);
 
 	for (i = 0; i < ISU_LAYER_NUM; i++) {
-		for (j = 0; j < ISU_LAYER_NUM; j++) {
+		for (j = 0; j < ISU_OFFSET_NUM; j++) {
 			wpf_info->k_matrix[i][j] = 0;
+			wpf_info->offset[i][j] = 0;
 		}
 	}
 	for (i = 0; i < ISU_LAYER_NUM; i++) {
-		for (j = 0; j < ISU_OFFSET_NUM; j++) {
-			wpf_info->offset[i][j] = 0;
-			wpf_info->clip[i][j] = 0xFF00;
-		}
+			wpf_info->clip[i][0] = 0;
+			wpf_info->clip[i][1] = 0xFF;
 	}
 
 	/* check color space conversion parameter */
-- 
2.7.4
